### ISSUE TITLE
bugs in vcf<->hap/sample conversion

### ISSUE THREAD

**Issue by dwuab** [samtools/bcftools#1385]
I'm using bcftools 1.11. Suppose I have a vcf file `test.vcf.gz`:
```
##fileformat=VCFv4.2
##FILTER=<ID=PASS,Description="All filters passed">
##INFO=<ID=END,Number=1,Type=Integer,Description="End position of the variant described in this record">
##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype">
##contig=<ID=1,length=2147483647>
##bcftools_convertVersion=1.11+htslib-1.11
##bcftools_convertCommand=convert --hapsample2vcf test2.hap; Date=Tue Jan 26 15:04:08 2021
#CHROM  POS     ID      REF     ALT     QUAL    FILTER  INFO    FORMAT  P-001
1       54490   .       G       A       .       .       .       GT      0|0
```
I converted it to hap/sample format by command `bcftools convert --hapsample test test.vcf.gz`, and then I got two files: `test.hap.gz` and `test.sample`. In particular, the content of `test.hap.gz` is
```
1 1:54490_G_A 54490 G A 0 0
```
which is already wrong since in bcftools's documentation (release 1.11), the first column should be the variant ID of format "CHR:POS_REF_ALT(_END)?".
First I wanted to convert "test.hap.gz,test.sample" back to vcf format by `bcftools convert --hapsample2vcf test`, and bcftools returned an error message `Could not read test.samples`, which is consistent because `bcftools convert --hapsample` produces a sample file with suffix `.sample`, not `.samples`.
Next I tried to bypass this inconsistency by using `bcftools convert --hapsample2vcf test.hap.gz,test.sample`, and bcftools returned an error message `Could not parse CHROM in CHROM:POS_REF_ALT id: 1 1:54490_G_A 54490 G A 0 0`, which is due to test.vcf.gz not properly transformed to hap/sample format earlier. 
Finally I modified the wrongly converted `.hap` file to `test2.hap`:
```
1:54490_G_A 1:54490_G_A 54490 G A 0 0
```, then `bcftools convert --hapsample2vcf test2.hap,test.sample` produce the what I had in test.vcf.gz.
To summarize:
1) The first column in `.hap` file produced by `bcftools convert --hapsample` is wrong.
2) `bcftools convert --hapsample` produces a sample file with suffix `.sample` while `bcftools convert --hapsample2vcf prefix` expects `prefix.samples`.
3) This is not exactly a bug. I know that there is no way to record whether a variant is phased in hap/sample format, but is it better to mark all variants as unphased (`*/*` instead of `*|*`) when converting from hap/sample format to vcf format?

**Comment by pd3** [comment]
Thank you for raising the issue, it is fixed now. Please let us know if you spot anything else.
